Makefile

```diff
@@ -247,134 +234,130 @@ instrumentation-test-lint: npm-installed
 	cd images/instrumentation/test && npm ci && npm run lint
 
 .PHONY: prometheus-crd-version-check
-prometheus-crd-version-check:
+prometheus-crd-version-check: ## Check whether all references to the PrometheusRule CRD are in sync.
 	@echo "-------------------------------- (verifying the Prometheus CRD version is in sync)"
 	./test-resources/bin/prometheus-crd-version-check.sh
 
 .PHONY: perses-crd-version-check
-perses-crd-version-check:
+perses-crd-version-check: ## Check whether all references to the PersesDashboard CRD are in sync.
 	@echo "-------------------------------- (verifying the Perses CRD version is in sync)"
 	./test-resources/bin/perses-crd-version-check.sh
 
-.PHONY: zig-fmt
-zig-fmt: zig-installed
-	zig fmt images/instrumentation/injector/src
-
 .PHONY: lint
-lint: golangci-lint helm-chart-lint shellcheck-lint zig-fmt-check instrumentation-test-lint perses-crd-version-check prometheus-crd-version-check
+lint: golangci-lint helm-chart-lint shellcheck-lint zig-fmt-check instrumentation-test-lint perses-crd-version-check prometheus-crd-version-check ## Run all static code analysis checks (Go, Helm, shell scripts, Zig, etc.).
 
 .PHONY: lint-fix
 lint-fix: golang-lint-fix zig-fmt
 
+PHONY: test-app-images
+test-app-images: \
+  test-app-image-dotnet \
+  test-app-image-jvm \
+  test-app-image-nodejs ## Build all test application container images. If IMAGE_PLATFORMS is set, it will be passed as --platform to the build.
+
+.PHONY: test-app-image-dotnet
+test-app-image-dotnet: ## Build the .NET test application.
+	@$(call build_container_image,$(TEST_APP_DOTNET_IMAGE_REPOSITORY),$(TEST_APP_DOTNET_IMAGE_TAG),test-resources/dotnet)
+
+.PHONY: test-app-image-jvm
+test-app-image-jvm: ## Build the JVM test application.
+	@$(call build_container_image,$(TEST_APP_JVM_IMAGE_REPOSITORY),$(TEST_APP_JVM_IMAGE_TAG),test-resources/jvm/spring-boot)
+
+.PHONY: test-app-image-nodejs
+test-app-image-nodejs: ## Build the Node.js test application.
+	@$(call build_container_image,$(TEST_APP_NODEJS_IMAGE_REPOSITORY),$(TEST_APP_NODEJS_IMAGE_TAG),test-resources/node.js/express)
+
+.PHONY: dash0-api-mock-image
+dash0-api-mock-image: ## Build the Dash0 API mock container image, which is used in end-to-end tests.
+	@$(call build_container_image,$(DASH0_API_MOCK_IMAGE_REPOSITORY),$(DASH0_API_MOCK_IMAGE_TAG),test/e2e/dash0-api-mock)
+
 ##@ Build
 
 .PHONY: build
-build: husky-setup-hooks manifests generate fmt vet ## Build manager binary.
+build: manifests generate fmt vet ## Build manager binary.
 	go build -o bin/manager cmd/main.go
 
-.PHONY: run
-run: manifests generate fmt vet ## Run a controller from your host.
-	go run ./cmd/main.go
```

Comment:
I removed this target for starting the operator directly on your local machine (without deploying it as an image to a cluster) as well, I think it is rather pointless to do that.